lib/rubygems/commands/update_command.rb

```diff
@@ -75,6 +75,13 @@ def check_latest_rubygems(version) # :nodoc:
     end
   end
 
+  def check_oldest_rubygems(version) # :nodoc:
+    if oldest_supported_version > version
+      alert_error "rubygems #{version} is not supported. The oldest supported version is #{oldest_supported_version}"
```

Comment:
Do we need to fail here? Isn't warning enough here?

Reply:
The related ticket proposed to prevent downgrades, warnings were not consider.

As I explained in the related ticket, when I downgraded to a very old rubygems, my ruby installation was completely messed up (rubygems is specially sensible to these issues since it's required by ruby by default). A warning wouln't have saved me from anything.

So, yeah, if we don't test against a specific rubygems version, I prefer to call it not supported and prevent downgrades.

Reply:
Also note that if you really want to downgrade to an ancient version and mess up your installation, you can still do it: downgrade to a rubygems version without this feature, and from there downgrade to wherever :)